test/unit/vm/disposable_test.dart

```diff
@@ -63,6 +26,60 @@ void main() {
       thing = new DisposableThing();
     });
 
+    group('getManagedTimer', () {
+      TimerHarness harness;
+      Timer timer;
+
+      setUp(() {
+        harness = new TimerHarness();
+        timer = thing.getManagedTimer(harness.duration, harness.getCallback());
+      });
+
+      test('should cancel timer if disposed before completion', () async {
+        expect(timer.isActive, isTrue);
+        await thing.dispose();
+        expect(await harness.didCancelTimer, isTrue);
+        expect(await harness.didCompleteTimer, isFalse);
+      });
+
+      test('disposing should have no effect on timer after it has completed',
+          () async {
+        await harness.didConclude;
+        expect(timer.isActive, isFalse);
+        await thing.dispose();
+        expect(await harness.didCancelTimer, isFalse);
+        expect(await harness.didCompleteTimer, isTrue);
+      });
+    });
+
+    group('getManagedPeriodicTimer', () {
+      TimerHarness harness;
+      Timer timer;
+
+      setUp(() {
+        harness = new TimerHarness();
+        timer = thing.getManagedPeriodicTimer(
+            harness.duration, harness.getPeriodicCallback());
+      });
+
+      test('should cancel timer if disposed before completion', () async {
+        expect(timer.isActive, isTrue);
+        await thing.dispose();
+        expect(await harness.didCancelTimer, isTrue);
+        expect(await harness.didCompleteTimer, isFalse);
+      });
+
+      test('disposing should have no effect on timer after it has completed',
```

Comment:
I updated the test description to indicate that when a consumer cancels the timer a dispose has no effect.